taxcalc/behavior.py

```diff
@@ -407,6 +480,179 @@ def _validate_parameter_values(self):
+                            scalar = False  # parameter value is a list
+                        else:
+                            scalar = True  # parameter value is a scalar
+                            pvalue = [pvalue]  # make scalar a single-item list
+                        for idx in range(0, len(pvalue)):
+                            if scalar:
+                                pname = name
+                            else:
+                                pname = '{}_{}'.format(name, idx)
+                            pvalue_boolean = (
+                                isinstance(pvalue[idx], bool) or
+                                (isinstance(pvalue[idx], int) and
+                                 (pvalue[idx] == 0 or pvalue[idx] == 1)) or
+                                (isinstance(pvalue[idx], float) and
+                                 (pvalue[idx] == 0.0 or pvalue[idx] == 1.0))
+                            )
+                            if bool_type:
+                                if not pvalue_boolean:
+                                    msg = '{} {} value {} is not boolean'
+                                    self.behavior_errors += (
+                                        'ERROR: ' +
+                                        msg.format(year, pname, pvalue[idx]) +
+                                        '\n'
+                                    )
+                            elif int_type:
+                                if not isinstance(pvalue[idx], int):
+                                    msg = '{} {} value {} is not integer'
+                                    self.behavior_errors += (
+                                        'ERROR: ' +
+                                        msg.format(year, pname, pvalue[idx]) +
+                                        '\n'
+                                    )
+                            else:  # param is neither bool_type nor int_type
+
+                                ################################
+                                # additional Behavior specific logic
+                                # isinstance(True, (float, int)) is True--
+                                # catches case where value is given as bool
+                                # but we are expecting an integer or float
+                                # i.e. boolean_value = integer_value = False
+                                ################################
+
+                                if (not isinstance(pvalue[idx], (float, int)) or
+                                    isinstance(pvalue[idx], bool)):
+                                    msg = '{} {} value {} is not a number'
+                                    self.behavior_errors += (
+                                        'ERROR: ' +
+                                        msg.format(year, pname, pvalue[idx]) +
+                                        '\n'
+                                    )
```

Comment:
The test case:
```
# test_behavior.py
def test_validate_param_names_types_errors():
...
+    behv1 = Behavior()
+    specs1 = {2019: {'_BE_inc': [True]}} # _BE_inc should be a float (int_type = boolean_type = False)
+    with pytest.raises(ValueError):
+        behv1.update_behavior(specs1) 
```
did not throw a `ValueError` before the second part of the conditional statement was added. See [policy.py](https://github.com/open-source-economics/Tax-Calculator/blob/master/taxcalc/policy.py#L482-L488) for original conditional statement.